example/main.rs

```diff
@@ -165,13 +171,28 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
                 matches.subcommand_matches(section_name).is_some()
             });
 
-    let key = matches.value_of("auth-key").unwrap();
-    let email = matches.value_of("email").unwrap();
+    let email = matches.value_of("email");
+    let key = matches.value_of("auth-key");
+    let token = matches.value_of("auth-token");
 
-    let api_client = HttpApiClient::new(Credentials::User {
-        key: key.to_string(),
-        email: email.to_string(),
-    });
+    // Create bogus default api_client; this will be overwritten with a client
+    // that uses a key or token.
+    let mut credentials = Credentials::Default;
```

Comment:
I don't think we need to add a special `Default` variant just for this assignment. Instead, we can assign `credentials` to the result of an `if`, like this:

```rust
let credentials = if let Some(key) = key {
    Credentials::UserAuthKey {
        email: email.unwrap().to_string(),
        key: key.to_string(),
    }
} else if let Some(token) = token {
    Credentials::UserAuthToken {
        token: token.to_string(),
    }
} else {
    panic!("either token or key must be specified")
}
```

Reply:
Ah! I'll try this. I tried using other error handling approaches (namely, failure crate) but the compiler was unhappy with me; just using `panic!` should work though!

Reply:
Great. I'd never put a `panic!` in the library itself, but in the example binary it's totally fine.